tools/espressif/btdecode.sh

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env bash
+############################################################################
+# tools/espressif/btdecode.sh
```

Comment:
it's a general tool, let's move out of espressif folder

Reply:
Do you mean to make it available for all chips? I'm not against it...

Reply:
Done!